lucene/core/src/java/org/apache/lucene/index/IndexWriterConfig.java

```diff
@@ -482,6 +483,15 @@ public IndexWriterConfig setCommitOnClose(boolean commitOnClose) {
     return this;
   }
 
+  /**
+   * Sets read time attributes that will be loaded into SegmentReadState while
+   * opening the index
+   */
+  public IndexWriterConfig setReaderAttributes(Map<String, String> readerAttributes) {
+    this.readerAttributes = readerAttributes;
```

Comment:
this one should be an unmodifiable map, you should also make a copy for the attributes passed in.

Reply:
Good point. Will make the change. Thanks!